src/actions/fetchPosts.js

```diff
@@ -30,6 +30,21 @@ export function fetchPosts (opts) {
 
   payload.path += '?' + querystring
 
-  let cache = {id: cacheId, bucket: 'postsByQuery', limit, offset, array: true}
-  return {type: FETCH_POSTS, payload, meta: {cache}}
+  var meta = {cache: {id: cacheId, bucket: 'postsByQuery', limit, offset, array: true}}
+
+  return {type: FETCH_POSTS, payload, meta}
+}
+
+export function checkFreshness (subject, id, posts, query = {}) {
```

Comment:
not sure if this belongs here. But they have a lot of overlap, so it made sense to put them together. Maybe just need to change the filename?